internal/service/domain/tree/tree_test.go

```diff
@@ -0,0 +1,831 @@
+		treeClusterRepo := storageMock.NewMockTreeClusterRepository(t)
+		locator := service.NewMockGeoClusterLocator(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, treeClusterRepo, locator)
+
+		expectedTree := getTestTrees()[0]
+		expectedTree.TreeCluster = getTestTreeClusters()[0]
+		expectedError := errors.New("cluster update failed")
+
+		// Mock expectations
+		treeRepo.EXPECT().GetByID(ctx, expectedTree.ID).Return(expectedTree, nil)
+		treeRepo.EXPECT().Delete(ctx, expectedTree.ID).Return(nil)
+		locator.EXPECT().UpdateCluster(ctx, &expectedTree.TreeCluster.ID).Return(expectedError)
+
+		// When
+		err := svc.Delete(ctx, expectedTree.ID)
+
+		// Then
+		assert.Error(t, err)
+		assert.EqualError(t, err, handleError(expectedError).Error())
+	})
+	t.Run("should delete a tree without triggering cluster update when tree has no cluster", func(t *testing.T) {
+		// Given
+		treeRepo := storageMock.NewMockTreeRepository(t)
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		imageRepo := storageMock.NewMockImageRepository(t)
+		treeClusterRepo := storageMock.NewMockTreeClusterRepository(t)
+		locator := service.NewMockGeoClusterLocator(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, treeClusterRepo, locator)
+
+		expectedTree := getTestTrees()[0]
+		expectedTree.TreeCluster = nil // Tree has no cluster
+
+		// Mock expectations
+		treeRepo.EXPECT().GetByID(ctx, expectedTree.ID).Return(expectedTree, nil)
+		treeRepo.EXPECT().Delete(ctx, expectedTree.ID).Return(nil)
+
+		// Ensure UpdateCluster is not called
+		locator.AssertNotCalled(t, "UpdateCluster", ctx, mock.Anything)
+
+		// When
+		err := svc.Delete(ctx, expectedTree.ID)
+
+		// Then
+		assert.NoError(t, err)
+	})
+
+}
+func TestTreeService_Ready(t *testing.T) {
+	t.Run("should return true when treeRepo and sensorRepo are initialized", func(t *testing.T) {
+		// Given
+		treeRepo := storageMock.NewMockTreeRepository(t)
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, nil, nil, nil)
+
+		// When
+		result := svc.Ready()
+
+		// Then
+		assert.True(t, result)
+	})
+
+	t.Run("should return false when treeRepo is nil", func(t *testing.T) {
+		// Given
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+
+		svc := NewTreeService(nil, sensorRepo, nil, nil, nil)
+
+		// When
+		result := svc.Ready()
+
+		// Then
+		assert.False(t, result)
+	})
+
+	t.Run("should return false when sensorRepo is nil", func(t *testing.T) {
+		// Given
+		treeRepo := storageMock.NewMockTreeRepository(t)
+
+		svc := NewTreeService(treeRepo, nil, nil, nil, nil)
+
+		// When
+		result := svc.Ready()
+
+		// Then
+		assert.False(t, result)
+	})
+
+	t.Run("should return false when both treeRepo and sensorRepo are nil", func(t *testing.T) {
+		// Given
+		svc := NewTreeService(nil, nil, nil, nil, nil)
+
+		// When
+		result := svc.Ready()
+
+		// Then
+		assert.False(t, result)
+	})
+}
+func TestTreeService_Create(t *testing.T) {
```

Comment:
Can you move this test up? Let's use the same order of tests as the methods are in the `test
.go` file.